source/s3_default_meta_request.c

```diff
@@ -216,6 +216,14 @@ static int s_s3_meta_request_default_prepare_request(
         aws_s3_meta_request_read_body(meta_request, &request->request_body);
     }
 
+    struct aws_http_headers *initial_request_headers =
+        aws_http_message_get_headers(meta_request->initial_request_message);
+
+    if (client->compute_content_md5 == AWS_MR_CONTENT_MD5_ENABLED &&
+        !aws_http_headers_has(initial_request_headers, g_content_md5_header_name)) {
```

Comment:
So the default meta request is sort of a tricky case because it's a catch-all for all S3 requests that are not gets or put.  As it is right now, I think we'll add content-md5 headers for a lot of cases where we shouldn't.

I think we should add a flag to the default meta request for if it's going to add the content-md5 header or not, and have that be specified through the default_meta_request_new function.  Then I think we can basically just move the logic for setting the flag here: https://github.com/awslabs/aws-c-s3/blob/main/source/s3_client.c#L996